src/snap-points.ts

```diff
@@ -93,34 +91,40 @@ export const createSnapshotForNetwork = async (network: Network) => {
     programId
   );
 
-  const previousConfig: IConfig = JSON.parse(
-    fs.readFileSync(configFileName, "utf-8")
+  const previousPools: Record<string, string | undefined> = JSON.parse(
+    fs.readFileSync(poolsFileName, "utf-8")
   );
 
-  const { lastTxHash } = previousConfig;
-
-  const refAddresses = PROMOTED_POOLS.map(
-    (pool) => market.getEventOptAccount(pool).address
+  const eventsObject: Record<string, IPositions> = JSON.parse(
+    fs.readFileSync(eventsSnapFilename, "utf-8")
   );
 
-  const sigArrays = await Promise.all(
-    refAddresses.map((refAddr) =>
-      retryOperation(fetchAllSignatures(connection, refAddr, lastTxHash))
+  const newPoolsFile = {};
+  const sigs = (
+    await Promise.all(
+      PROMOTED_POOLS.map((pool) => {
+        const refAddr = market.getEventOptAccount(pool).address;
+        const previousTxHash = previousPools[pool.toString()] ?? undefined;
```

Comment:
We dont want to pass `undefined` with undefined you are going to fetch all transaction from the mainnet deploy of the program. We should use transaction hash from which we start counting